### ISSUE TITLE
How to use random search in tuning hyperparameters?

### ISSUE THREAD

**Issue by Jiang-Lu-CAS** [deepchem/deepchem#1860]
I just found two methods in the hyper file (grid_search.py, gaussian_process.py). How can I use random search method? 
What's more, I found a  RandomizedSearchCV method in sklearn. 
`from sklearn.model_selection import RandomizedSearchCV`
Could I use this method to tune the GraphConvModel model in deepchem?

**Comment by rbharath** [comment]
@Jiang-Lu-CAS We should definitely have better support for random search :). I'll try to upstream this into head.

In the meanwhile, you can implement a rough random research sweep manually. Something like follows:
```
results = {}
for lr in np.random.choice([0.1, 0.01, 0.001, 0.0001]):
  for num_layers in np.random.choice([1, 2, 3, 4]):
    for layer_size in np.random.choice([16, 32, 64]):
      model = dc.models.GraphConvModel(learning_rate=lr, graph_conv_layers=[layer_size]*num_layers)
      model.fit(train_dataset, nb_epoch=50)
      score = model.evaluate(valid_dataset, [metric])
      results[(lr, num_layers, layer_size)] = score
```

**Comment by Jiang-Lu-CAS** [comment]
> @Jiang-Lu-CAS We should definitely have better support for random search :). I'll try to upstream this into head.
> In the meanwhile, you can implement a rough random research sweep manually. Something like follows:
> results = {}
> for lr in np.random.choice([0.1, 0.01, 0.001, 0.0001]):
>   for num_layers in np.random.choice([1, 2, 3, 4]):
>     for layer_size in np.random.choice([16, 32, 64]):
>       model = dc.models.GraphConvModel(learning_rate=lr, graph_conv_layers=[layer_size]*num_layers)
>       model.fit(train_dataset, nb_epoch=50)
>       score = model.evaluate(valid_dataset, [metric])
>       results[(lr, num_layers, layer_size)] = score

Thank you for your suggestion. Looking forward your random search support in deepchem :).
